src/help/index.ts

```diff
@@ -90,6 +90,8 @@ export class Help extends HelpBase {
   }
 
   public async showHelp(argv: string[]) {
+    argv = argv.filter(arg => !getHelpFlagAdditions(this.config).includes(arg))
```

Comment:
Moved the stripping of help flags to showHelp from main.ts.

Other function call via showHelp expect the help flags to be removed.